src/isolate.rs

```diff
@@ -66,6 +68,35 @@ impl IsolateState {
     let tx = maybe_tx.unwrap();
     tx.send((req_id, buf)).expect("tx.send error");
   }
+
+  fn update_metrics(&self, bytes_recv: u64, bytes_sent: u64) {
+    let mut g = self.metrics.lock().unwrap();
+    let metrics = g.deref_mut();
```

Comment:
You don't normally need to call `deref_mut`, since Rust automatically inserts it. Does it work if you do the following?

```rust
let mut metrics = self.metrics.lock().unwrap();
metrics.update(bytes_recv, bytes_sent);
```

Reply:
Thanks, worked like a charm!